### PR TITLE
Rework a bit rhol_crc

### PR DESCRIPTION
**Pull request by cjeanner**
There were multiple issues in the role:
- no version check, meaning it would consume whatever version it finds
  on the disk instead of ensuring it's consuming the target version
- lack of molecule tests for some specific and sensitive parts

This change also moves the "crc" binary to a more conventional location
(~/bin) to allow an easier discovery, and avoiding potential conflicts.

It also removes some wrong/useless parameters that could lead to weird
situations regarding rights, users and so on.

This PR also bumps the CRC version in both role and molecule (this
creates a conflict with another PR, #2102).

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role

Depends-On: https://github.com/openstack-k8s-operators/ci-framework/pull/2134

### CODE REVIEW

**Comment by cjeanner** [comment]
recheck

**Comment by cjeanner** [comment]
Fixes for pre-commit are in #2134

File: zuul.d/edpm.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@
     vars:
       crc_parameters: "--memory 32000 --disk-size 240 --cpus 12"
       cifmw_manage_secrets_pullsecret_content: '{}'
+      cifmw_rhol_crc_binary_folder: "/usr/local/bin"
```
Comment (cjeanner) [inline zuul.d/edpm.yaml:20]: this is needed in order to ensure we're using the pre-installed CRC. Until now, it was fetching its own one, meaning we could have some weird situations.

File: ci/config/molecule.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@
     nodeset: centos-9-crc-2-39-0-xl
 - job:
     name: cifmw-molecule-rhol_crc
-    nodeset: centos-9-crc-2-36-0-xxl #TODO(Lewis): Bump to 2-39 once OSPCIX-376 is resolved
```
Comment (cjeanner) [inline ci/config/molecule.yaml:18]: @lewisdenny so yep, it's done in here - up to you if you want me to split.

File: roles/rhol_crc/defaults/main.yml [hunk 3]
```diff
@@ -18,8 +18,6 @@
 # All variables intended for modification should be placed in this file.
 # All variables within this role should have a prefix of "cifmw_rhol_crc"
 
-# install_yamls root location
-cifmw_rhol_crc_installyamls: "{{ cifmw_installyamls_repos | default('../..') }}"
```
Comment (cjeanner) [inline roles/rhol_crc/defaults/main.yml:22]: not used - removed.

File: roles/rhol_crc/tasks/configuration.yml [hunk 4]
```diff
@@ -16,21 +16,42 @@
...
-          {{ cifmw_rhol_crc_binary }} config set
-          {{ item.key }} {{ item.value }} 2>&1 |
-          tee -a {{ cifmw_rhol_crc_basedir }}/logs/crc-config.log
-      loop: >-
+  vars:
+    _patches: >-
+      {{
+        hostvars[inventory_hostname] |
+        dict2items |
+        selectattr('key', 'match', 'cifmw_rhol_crc_config_patch_.*') |
```
Comment (lewisdenny) [inline roles/rhol_crc/tasks/configuration.yml:24]: I like this but I don't see where `cifmw_rhol_crc_config_patch_*` is documented.

File: roles/rhol_crc/README.md [hunk 5]
```diff
@@ -11,7 +11,7 @@ Become - required for the tasks in `sudoers_grant.yml` and `sudoers_revoke.yml`
...
 * `cifmw_rhol_crc_use_installyamls`: (Boolean) Mimic some steps of the install_yamls way to deploy CRC. Defaults to `False`.
 * `cifmw_rhol_crc_dryrun`: (Boolean) Toggle the `ci_make` `dry_run` parameter. Defaults to `False`.
-* `cifmw_rhol_crc_config`: (Dict) This structure is merged with the `cifmw_rhol_crc_config_defaults` dictionary. We can add extra properties or override the existing one using this parameter. We can know the parameters that can be used with the output of the `crc config --help` command. Defaults to `{}`.
+* `cifmw_rhol_crc_config`: (Dict) This structure is merged with the `cifmw_rhol_crc_config_defaults` dictionary. We can add extra properties or override the existing one using this parameter. We can know the parameters that can be used with the output of the `crc config --help` command. Defaults to `{}`. You can also `patch` configuration by using `cifmw_rhol_crc_config_path_*` pattern and providing the same kind of dict.
 * `cifmw_rhol_crc_version`: (String) RHOL/CRC binary version we want to use, set to `latest` to deploy latest CRC release. Default: `2.36.0`.
```
Comment (lewisdenny) [inline roles/rhol_crc/README.md:15]: 2.39.0

File: roles/rhol_crc/tasks/configuration.yml (hunk 4 above)
Comment (lewisdenny) [reply roles/rhol_crc/tasks/configuration.yml:24]: Ah I see, there is a typo!

File: roles/rhol_crc/README.md [hunk 6]
```diff
@@ -11,7 +11,7 @@ Become - required for the tasks in `sudoers_grant.yml` and `sudoers_revoke.yml`
 * `cifmw_rhol_crc_basedir`: (String) Directory where we will have the RHOL/CRC binary and the configuration (e.g. `artifacts/.rhol_crc_pull_secret.txt`). Default to `cifmw_basedir` which defaults to `~/ci-framework-data`.
 * `cifmw_rhol_crc_use_installyamls`: (Boolean) Mimic some steps of the install_yamls way to deploy CRC. Defaults to `False`.
 * `cifmw_rhol_crc_dryrun`: (Boolean) Toggle the `ci_make` `dry_run` parameter. Defaults to `False`.
-* `cifmw_rhol_crc_config`: (Dict) This structure is merged with the `cifmw_rhol_crc_config_defaults` dictionary. We can add extra properties or override the existing one using this parameter. We can know the parameters that can be used with the output of the `crc config --help` command. Defaults to `{}`.
+* `cifmw_rhol_crc_config`: (Dict) This structure is merged with the `cifmw_rhol_crc_config_defaults` dictionary. We can add extra properties or override the existing one using this parameter. We can know the parameters that can be used with the output of the `crc config --help` command. Defaults to `{}`. You can also `patch` configuration by using `cifmw_rhol_crc_config_path_*` pattern and providing the same kind of dict.
```
Comment (lewisdenny) [inline roles/rhol_crc/README.md:14]: cifmw_rhol_crc_config_path_ should be cifmw_rhol_crc_config_patch_

**Review by lewisdenny — CHANGES_REQUESTED** [review]
Thank you Cédric, big improvement! Have left some comments.

File: roles/rhol_crc/tasks/configuration.yml (hunk 4 above)
Comment (lewisdenny) [reply roles/rhol_crc/tasks/configuration.yml:24]: Ignore this one :)

**Comment by cjeanner** [comment]
Comment addressed.

**Review by lewisdenny — APPROVED** [review]
/approve

**Comment by cjeanner** [comment]
/hold

Some more testing via DS TP is requested.

**Pull request merged**